gitreceived/gitreceived.go

```diff
@@ -302,19 +321,70 @@ func exitStatus(err error) (exitStatusMsg, error) {
 
 var cacheMtx sync.Mutex
 
-func ensureCacheRepo(path string) error {
+func ensureCacheRepo(tempDir, path string) error {
 	cacheMtx.Lock()
 	defer cacheMtx.Unlock()
 
-	cachePath := *repoPath + "/" + path
-	if _, err := os.Stat(cachePath); os.IsNotExist(err) {
-		os.MkdirAll(cachePath, 0755)
+	cachePath := tempDir + "/" + path
+
+	if !*useBlobstore {
```

Comment:
Should probably split this into two functions since the blobstore path doesn't need a mutex.